internal/service/encryptionatrestprivateendpoint/resource_schema.go

```diff
@@ -0,0 +1,90 @@
+package encryptionatrestprivateendpoint
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+)
+
+func ResourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"cloud_provider": schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "Human-readable label that identifies the cloud provider for the Encryption At Rest private endpoint.",
+				MarkdownDescription: "Human-readable label that identifies the cloud provider for the Encryption At Rest private endpoint.",
+			},
+			"endpoint_id": schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "Unique 24-hexadecimal digit string that identifies the private endpoint.",
+				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies the private endpoint.",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(24, 24),
+					stringvalidator.RegexMatches(regexp.MustCompile("^([a-f0-9]{24})$"), ""),
+				},
+			},
+			"error_message": schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "Error message for failures associated with the Encryption At Rest private endpoint.",
+				MarkdownDescription: "Error message for failures associated with the Encryption At Rest private endpoint.",
+			},
+			"project_id": schema.StringAttribute{
+				Optional:            true,
```

Comment:
I would double-check the Optional/Computed for all these. Looks like everyone is optional-computed. I would expected `region_name`++ to be `required`?

Reply:
right